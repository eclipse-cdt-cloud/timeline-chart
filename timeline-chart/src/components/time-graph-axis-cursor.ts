import { TimeGraphComponent, TimeGraphElementPosition, TimeGraphComponentOptions } from "./time-graph-component";

export interface TimeGraphAxisCursorOptions extends TimeGraphComponentOptions {
    position: TimeGraphElementPosition
    color: number
}

export class TimeGraphAxisCursor extends TimeGraphComponent<null> {

    constructor(protected _options: TimeGraphAxisCursorOptions) {
        super('cursor');
    }

    render(): void {
        const { position, color } = this._options;
        this._displayObject.beginFill(color);
        this._displayObject.moveTo(position.x, position.y);
        this._displayObject.lineTo(position.x - 5, position.y - 5);
        this._displayObject.lineTo(position.x + 5, position.y - 5);
        this._displayObject.lineTo(position.x, position.y);
        this._displayObject.endFill();
    }

}