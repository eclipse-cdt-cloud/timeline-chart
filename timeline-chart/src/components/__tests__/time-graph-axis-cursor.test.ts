import { TimeGraphAxisCursor } from '../time-graph-axis-cursor';

describe ('TimeGraphAxisCursor', () => {
    const options = {
        position: { x: 0, y: 1},
        color: 111
    }
    const component = new TimeGraphAxisCursor(options);

    afterAll(() => {
        component.destroy();
    });

    it ('Renders', () => {
        expect(component).toBeTruthy();
    })

    it ('Matches snapshot', () => {
        expect(component).toMatchSnapshot();
    })
})
