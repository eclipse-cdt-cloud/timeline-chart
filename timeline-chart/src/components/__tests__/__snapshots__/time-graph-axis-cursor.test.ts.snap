// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeGraphAxisCursor Matches snapshot 1`] = `
TimeGraphAxisCursor {
  "_displayObject": _Graphics2 {
    "_bounds": Bounds {
      "maxX": -Infinity,
      "maxY": -Infinity,
      "minX": Infinity,
      "minY": Infinity,
      "rect": null,
      "updateID": -1,
    },
    "_boundsID": 0,
    "_boundsRect": null,
    "_destroyed": false,
    "_enabledFilters": null,
    "_events": {},
    "_eventsCount": 0,
    "_fillStyle": FillStyle {
      "alpha": 1,
      "color": 16777215,
      "matrix": null,
      "texture": Texture {
        "_events": {},
        "_eventsCount": 0,
        "_frame": Rectangle {
          "height": 16,
          "type": 1,
          "width": 16,
          "x": 0,
          "y": 0,
        },
        "_rotate": 0,
        "_updateID": 1,
        "_uvs": TextureUvs {
          "uvsFloat32": Float32Array [
            0,
            0,
            1,
            0,
            1,
            1,
            0,
            1,
          ],
          "x0": 0,
          "x1": 1,
          "x2": 1,
          "x3": 0,
          "y0": 0,
          "y1": 0,
          "y2": 1,
          "y3": 1,
        },
        "baseTexture": _BaseTexture2 {
          "_batchEnabled": 0,
          "_batchLocation": 0,
          "_events": {
            "update": EE {
              "context": [Circular],
              "fn": [Function],
              "once": false,
            },
          },
          "_eventsCount": 1,
          "_glTextures": {},
          "_mipmap": 1,
          "_scaleMode": 1,
          "_wrapMode": 33071,
          "alphaMode": 1,
          "anisotropicLevel": 0,
          "cacheId": "pixiid_1",
          "destroy": [Function],
          "destroyed": false,
          "dirtyId": 0,
          "dirtyStyleId": 0,
          "emit": [Function],
          "format": 6408,
          "height": 16,
          "isPowerOfTwo": true,
          "on": [Function],
          "once": [Function],
          "parentTextureArray": null,
          "resolution": 1,
          "resource": CanvasResource {
            "_height": 16,
            "_width": 16,
            "destroyed": false,
            "internal": true,
            "noSubImage": false,
            "onError": Runner {
              "_aliasCount": 0,
              "_name": "onError",
              "items": [
                [Circular],
              ],
            },
            "onResize": Runner {
              "_aliasCount": 0,
              "_name": "setRealSize",
              "items": [
                [Circular],
              ],
            },
            "onUpdate": Runner {
              "_aliasCount": 0,
              "_name": "update",
              "items": [
                [Circular],
              ],
            },
            "source": <canvas
              height="16"
              width="16"
            />,
          },
          "target": 3553,
          "textureCacheIds": [
            "pixiid_1",
          ],
          "touched": 0,
          "type": 5121,
          "uid": 2,
          "valid": true,
          "width": 16,
        },
        "defaultAnchor": Point {
          "x": 0,
          "y": 0,
        },
        "defaultBorders": undefined,
        "destroy": [Function],
        "destroyed": false,
        "emit": [Function],
        "noFrame": true,
        "on": [Function],
        "once": [Function],
        "orig": Rectangle {
          "height": 16,
          "type": 1,
          "width": 16,
          "x": 0,
          "y": 0,
        },
        "textureCacheIds": [],
        "trim": undefined,
        "uvMatrix": null,
        "valid": true,
      },
      "visible": false,
    },
    "_geometry": _GraphicsGeometry2 {
      "_bounds": Bounds {
        "maxX": -Infinity,
        "maxY": -Infinity,
        "minX": Infinity,
        "minY": Infinity,
        "rect": null,
        "updateID": -1,
      },
      "_buffer": Buffer {
        "_glBuffers": {},
        "_updateID": 0,
        "data": Float32Array [
          0,
        ],
        "disposeRunner": Runner {
          "_aliasCount": 0,
          "_name": "disposeBuffer",
          "items": [],
        },
        "id": 0,
        "static": false,
        "type": 34962,
      },
      "_indexBuffer": Buffer {
        "_glBuffers": {},
        "_updateID": 0,
        "data": Float32Array [
          0,
        ],
        "disposeRunner": Runner {
          "_aliasCount": 0,
          "_name": "disposeBuffer",
          "items": [],
        },
        "id": 1,
        "static": false,
        "type": 34963,
      },
      "attributes": {
        "aColor": Attribute {
          "buffer": 0,
          "divisor": 1,
          "instance": false,
          "normalized": true,
          "size": 4,
          "start": undefined,
          "stride": undefined,
          "type": 5121,
        },
        "aTextureCoord": Attribute {
          "buffer": 0,
          "divisor": 1,
          "instance": false,
          "normalized": false,
          "size": 2,
          "start": undefined,
          "stride": undefined,
          "type": 5126,
        },
        "aTextureId": Attribute {
          "buffer": 0,
          "divisor": 1,
          "instance": false,
          "normalized": true,
          "size": 1,
          "start": undefined,
          "stride": undefined,
          "type": 5126,
        },
        "aVertexPosition": Attribute {
          "buffer": 0,
          "divisor": 1,
          "instance": false,
          "normalized": false,
          "size": 2,
          "start": undefined,
          "stride": undefined,
          "type": 5126,
        },
      },
      "batchDirty": -1,
      "batchable": false,
      "batches": [],
      "boundsDirty": -1,
      "boundsPadding": 0,
      "buffers": [
        Buffer {
          "_glBuffers": {},
          "_updateID": 0,
          "data": Float32Array [
            0,
          ],
          "disposeRunner": Runner {
            "_aliasCount": 0,
            "_name": "disposeBuffer",
            "items": [],
          },
          "id": 0,
          "static": false,
          "type": 34962,
        },
        Buffer {
          "_glBuffers": {},
          "_updateID": 0,
          "data": Float32Array [
            0,
          ],
          "disposeRunner": Runner {
            "_aliasCount": 0,
            "_name": "disposeBuffer",
            "items": [],
          },
          "id": 1,
          "static": false,
          "type": 34963,
        },
      ],
      "cacheDirty": -1,
      "clearDirty": 0,
      "closePointEps": 0.0001,
      "colors": [],
      "dirty": 0,
      "disposeRunner": Runner {
        "_aliasCount": 0,
        "_name": "disposeGeometry",
        "items": [],
      },
      "drawCalls": [],
      "glVertexArrayObjects": {},
      "graphicsData": [],
      "id": 0,
      "indexBuffer": Buffer {
        "_glBuffers": {},
        "_updateID": 0,
        "data": Float32Array [
          0,
        ],
        "disposeRunner": Runner {
          "_aliasCount": 0,
          "_name": "disposeBuffer",
          "items": [],
        },
        "id": 1,
        "static": false,
        "type": 34963,
      },
      "indices": [],
      "indicesUint16": null,
      "instanceCount": 1,
      "instanced": false,
      "points": [],
      "refCount": 1,
      "shapeIndex": 0,
      "textureIds": [],
      "uvs": [],
      "uvsFloat32": null,
    },
    "_holeMode": false,
    "_lastSortedIndex": 0,
    "_lineStyle": LineStyle {
      "alignment": 0.5,
      "alpha": 1,
      "cap": "butt",
      "color": 0,
      "join": "miter",
      "matrix": null,
      "miterLimit": 10,
      "native": false,
      "texture": Texture {
        "_events": {},
        "_eventsCount": 0,
        "_frame": Rectangle {
          "height": 16,
          "type": 1,
          "width": 16,
          "x": 0,
          "y": 0,
        },
        "_rotate": 0,
        "_updateID": 1,
        "_uvs": TextureUvs {
          "uvsFloat32": Float32Array [
            0,
            0,
            1,
            0,
            1,
            1,
            0,
            1,
          ],
          "x0": 0,
          "x1": 1,
          "x2": 1,
          "x3": 0,
          "y0": 0,
          "y1": 0,
          "y2": 1,
          "y3": 1,
        },
        "baseTexture": _BaseTexture2 {
          "_batchEnabled": 0,
          "_batchLocation": 0,
          "_events": {
            "update": EE {
              "context": [Circular],
              "fn": [Function],
              "once": false,
            },
          },
          "_eventsCount": 1,
          "_glTextures": {},
          "_mipmap": 1,
          "_scaleMode": 1,
          "_wrapMode": 33071,
          "alphaMode": 1,
          "anisotropicLevel": 0,
          "cacheId": "pixiid_1",
          "destroy": [Function],
          "destroyed": false,
          "dirtyId": 0,
          "dirtyStyleId": 0,
          "emit": [Function],
          "format": 6408,
          "height": 16,
          "isPowerOfTwo": true,
          "on": [Function],
          "once": [Function],
          "parentTextureArray": null,
          "resolution": 1,
          "resource": CanvasResource {
            "_height": 16,
            "_width": 16,
            "destroyed": false,
            "internal": true,
            "noSubImage": false,
            "onError": Runner {
              "_aliasCount": 0,
              "_name": "onError",
              "items": [
                [Circular],
              ],
            },
            "onResize": Runner {
              "_aliasCount": 0,
              "_name": "setRealSize",
              "items": [
                [Circular],
              ],
            },
            "onUpdate": Runner {
              "_aliasCount": 0,
              "_name": "update",
              "items": [
                [Circular],
              ],
            },
            "source": <canvas
              height="16"
              width="16"
            />,
          },
          "target": 3553,
          "textureCacheIds": [
            "pixiid_1",
          ],
          "touched": 0,
          "type": 5121,
          "uid": 2,
          "valid": true,
          "width": 16,
        },
        "defaultAnchor": Point {
          "x": 0,
          "y": 0,
        },
        "defaultBorders": undefined,
        "destroy": [Function],
        "destroyed": false,
        "emit": [Function],
        "noFrame": true,
        "on": [Function],
        "once": [Function],
        "orig": Rectangle {
          "height": 16,
          "type": 1,
          "width": 16,
          "x": 0,
          "y": 0,
        },
        "textureCacheIds": [],
        "trim": undefined,
        "uvMatrix": null,
        "valid": true,
      },
      "visible": false,
      "width": 0,
    },
    "_localBounds": null,
    "_localBoundsRect": null,
    "_mask": null,
    "_maskRefCount": 0,
    "_matrix": null,
    "_tintColor": _Color2 {
      "_components": Float32Array [
        1,
        1,
        1,
        1,
      ],
      "_int": 16777215,
      "_value": 16777215,
    },
    "_transformID": -1,
    "_zIndex": 0,
    "alpha": 1,
    "batchDirty": -1,
    "batchTint": -1,
    "batches": [],
    "children": [],
    "cullArea": null,
    "cullable": false,
    "currentPath": null,
    "filterArea": null,
    "filters": null,
    "isMask": false,
    "isSprite": false,
    "parent": null,
    "pluginName": "batch",
    "renderable": true,
    "shader": null,
    "sortDirty": false,
    "sortableChildren": false,
    "state": State {
      "_blendMode": 0,
      "_polygonOffset": 0,
      "data": 33,
    },
    "tempDisplayObjectParent": null,
    "transform": _Transform {
      "_currentLocalID": 0,
      "_cx": 1,
      "_cy": 0,
      "_localID": 0,
      "_parentID": 0,
      "_rotation": 0,
      "_sx": 0,
      "_sy": 1,
      "_worldID": 0,
      "localTransform": Matrix {
        "a": 1,
        "array": null,
        "b": 0,
        "c": 0,
        "d": 1,
        "tx": 0,
        "ty": 0,
      },
      "pivot": ObservablePoint {
        "_x": 0,
        "_y": 0,
        "cb": [Function],
        "scope": [Circular],
      },
      "position": ObservablePoint {
        "_x": 0,
        "_y": 0,
        "cb": [Function],
        "scope": [Circular],
      },
      "scale": ObservablePoint {
        "_x": 1,
        "_y": 1,
        "cb": [Function],
        "scope": [Circular],
      },
      "skew": ObservablePoint {
        "_x": 0,
        "_y": 0,
        "cb": [Function],
        "scope": [Circular],
      },
      "worldTransform": Matrix {
        "a": 1,
        "array": null,
        "b": 0,
        "c": 0,
        "d": 1,
        "tx": 0,
        "ty": 0,
      },
    },
    "vertexData": null,
    "visible": true,
    "worldAlpha": 1,
  },
  "_id": "cursor",
  "_options": {
    "color": 111,
    "position": {
      "x": 0,
      "y": 1,
    },
  },
}
`;
